NationStatesAPIBot/NationStatesAPIBot/Services/RecruitmentService.cs

```diff
@@ -110,14 +94,27 @@ public async Task<List<Nation>> GetRecruitableNationsAsync(int number)
 
         }
 
+        private async Task<bool> IsNationRecruitableAsync(Nation nation, EventId id)
+        {
+            if (nation != null)
+            {
+                if (!await DoesNationFitCriteriaAsync(nation) || !await WouldReceiveTelegram(nation.Name))
+                {
+                    _logger.LogDebug(id, LogMessageBuilder.Build(id, $"{nation.Name} does not fit criteria and is therefore skipped"));
+                    await NationManager.SetNationStatusToAsync(nation, "skipped");
+                    return false;
+                }
+                return true;
+            }
+            return false;
+        }
+
         private async Task<bool> DoesNationFitCriteriaAsync(Nation nation)
         {
             if (_config.CriteriaCheckOnNations)
             {
-                string pattern = @"(^[0-9]+?|[0-9]+?$)";
-                var name = nation.Name;
-                var res = !Regex.IsMatch(name, pattern);
-
+                var res = !nation.Name.Any(c => char.IsDigit(c)) && !(nation.Name.Count(c => c == nation.Name[0]) == nation.Name.Length);
```

Comment:
I think this can be written as:
`!nation.Name.Any(char.IsDigit) && nation.Name.Count(c => c == nation.Name[0]) != nation.Name.Length;`

Reply:
That's true